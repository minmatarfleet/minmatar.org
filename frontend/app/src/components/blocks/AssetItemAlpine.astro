---
import { i18n } from '@helpers/i18n'
const { t } = i18n(Astro.url)

import FixedFluid from '@components/compositions/FixedFluid.astro';
import Flexblock from '@components/compositions/Flexblock.astro';
import FlexInline from '@components/compositions/FlexInline.astro';

import VerticalCenter from '@components/blocks/VerticalCenter.astro';
---

<div>
    <template x-if="assets_style === 'list'">
        <FixedFluid
            width='64px'
            class="[ asset-item ]"
            {...Astro.props}
        >
            <picture>
                <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt={'`${asset.name} '+t('icon').toLowerCase()+'`'} />
            </picture>
            <FlexInline justification='space-between'>
                <VerticalCenter>
                    <Flexblock gap='0' class="[ w-full ]">
                        <span x-text={'`${asset.count} x ${asset.name}`'} />
                    </Flexblock>
                </VerticalCenter>
            </FlexInline>
        </FixedFluid>
    </template>

    <template x-if="assets_style === 'icons'">
        <Flexblock class="[ asset-icon ]" centered={true} gap='var(--space-3xs)'>
            <picture>
                <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt={'`${asset.name} '+t('icon').toLowerCase()+'`'} />
            </picture>
            <small x-text={'`${asset.name}`'} />
        </Flexblock>
    </template>
</div>

<style lang="scss">
    .asset-icon {
        small {
            height: 40px;
            overflow: hidden;
            text-align: center;
        }
    }
</style>