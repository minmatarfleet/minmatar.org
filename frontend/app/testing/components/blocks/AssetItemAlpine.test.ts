import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AssetItemAlpine from "@components/blocks/AssetItemAlpine.astro";

test("AssetItemAlpine defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AssetItemAlpine, {});

  expect(result).toMatchSnapshot();
});
