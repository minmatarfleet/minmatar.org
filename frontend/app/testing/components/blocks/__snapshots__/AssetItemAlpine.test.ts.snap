// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetItemAlpine defaults 1`] = `"<div data-astro-cid-ekkdkgyu> <template x-if="assets_style === 'list'" data-astro-cid-ekkdkgyu> <div class="fixed-fluid [ asset-item ]" data-astro-cid-ekkdkgyu="true" data-mod data-astro-cid-34go4nvk style="--fixed_fluid_gap: var(--space-s);--fixed_fluid_width: 64px;--fixed_fluid_breakpoint: 50%;">  <picture data-astro-cid-ekkdkgyu> <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt="\`\${asset.name} icon\`" data-astro-cid-ekkdkgyu> </picture> <div class="flex-inline" data-astro-cid-ekkdkgyu="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-s);--flex_inline_justification: space-between;">  <div class="!flex !items-center" data-astro-cid-ekkdkgyu="true">  <div class="flex-block [ w-full ] gap-0" data-astro-cid-ekkdkgyu="true" data-mod style="--flex_block_gap: 0;">  <span x-text="\`\${asset.count} x \${asset.name}\`" data-astro-cid-ekkdkgyu></span>  </div>   </div>  </div>    </div>   </template> <template x-if="assets_style === 'icons'" data-astro-cid-ekkdkgyu> <div class="flex-block [ asset-icon ] gap---space-3xs" data-astro-cid-ekkdkgyu="true" data-mod="centered" style="--flex_block_gap: var(--space-3xs);">  <picture data-astro-cid-ekkdkgyu> <img loading="lazy" x-bind:src="get_item_icon(asset.id)" width="64" height="64" x-bind:alt="\`\${asset.name} icon\`" data-astro-cid-ekkdkgyu> </picture> <small x-text="\`\${asset.name}\`" data-astro-cid-ekkdkgyu></small>  </div>  </template> </div> "`;
